### DISCUSSION TITLE
gpt is hilarious sometimes

### DISCUSSION THREAD

**Discussion by asmrtfm** [terrastruct/d2#1876] [Show and tell]
I asked chat gpt to give me some d2lang for a horizontal tree view of a project directory structure... got:

---

_Creating a horizontal tree-view script in D2Lang specifically for the <DIR> directory structure is not directly feasible because_

### _**> `D2Lang (Diablo II scripting language)`**_

_is primarily designed for creating mods and scripts within the context of the Diablo II game and does not have built-in capabilities for filesystem operations or generating directory trees._

However, for a general approach to creating a horizontal tree-view of a directory structure in a more commonly used scripting language like Python, you can use the following Python script as a starting point. This script will print the directory structure of <DIR> horizontally, with each subdirectory and its contents expanded below it.

```python
import os

def print_tree(startpath):
    for root, dirs, files in os.walk(startpath):
        level = root.replace(startpath, '').count(os.sep)
        indent = ' ' * 4 * (level)
        print('{}{}/'.format(indent, os.path.basename(root)))
        subindent = ' ' * 4 * (level + 1)
        for f in files:
            print('{}{}'.format(subindent, f))

startpath = <DIR>  # Adjust the path to your <DIR> directory
print_tree(startpath)
```

This script walks through the directory structure starting from <DIR> and prints each directory and file, indented to represent the structure. While this script provides a vertical tree view, modifying it to display a horizontal tree view would significantly increase its complexity due to the need for dynamic spacing and alignment adjustments based on the directory contents.

For a horizontal tree view, you might consider using existing command-line tools like `tree` in a Unix-like environment, which can give you a quick overview of the directory structure, although not in the exact format you described.

**Comment by alixander** [comment]
Our SEO arch nemesis, diablo 2!
